packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -238,6 +239,30 @@ export default class TabGeneral extends Component {
     }
   )
 
+  _getUrl = createSelector(this._getDiskGroups, diskGroups => childProps => {
+    const disks = []
+    const diskGroupsById = keyBy(diskGroups, 'id')
+
+    if (Array.isArray(childProps)) {
+      // "Others" section
+      const groups = pick(diskGroupsById, map(childProps, 'id'))
+      forEach(groups, ({ id, baseCopies = [], vdis, snapshots = [] }) => {
+        vdis === undefined
+          ? (disks[id] = { id })
+          : disks.push(...vdis, ...snapshots, ...baseCopies)
```

Comment:
For simplification, this approach is good but we loose performance on looping multiple times in collections. I suggest to construct the VDIs pattern instead.
Example:
```js
const vdisPattern = 'id:|'
vdis.forEach(({ id }) => {
  vdisPattern += id + ' '
})
```